pypesto/objective/history.py

```diff
@@ -1265,27 +1304,25 @@ def _check_file_id(self, file: str):
 
         Returns
         -------
-        file:
-            HDF5 file name.
         editable:
-            Boolean, whether this hdf5 file should be editable. Returns
-            false if the history is a loaded one to prevent overwriting.
-
+            Boolean, whether this hdf5 file should be editable.
+            Returns true if the file or the id entry does not exist yet.
         """
         try:
             with h5py.File(file, 'r') as f:
-                return file, (
-                    'history' not in f.keys() or self.id not in f['history']
-                )
-        except OSError:  # if the file is non-existent, return editable = True
-            return file, True
+                # editable if the id entry does not exist
```

Comment:
but what if the file exists and but is locked by another process? `h5py.File(file, 'r')` probably goes through fine but `h5py.File(file, 'w')` will error.

Reply:
changing to 'a' :+1: